### PR TITLE
Resource Mover MSI Access

### PR DESCRIPTION
**Pull request by punit1396**
Adding changes to provide MoveCollection MSI with required target key vault key and secret permissions. This is only done if the script is used with the new switch parameter. The move collection details are gathered by checking the resource links associated with the selected VMs.

### CODE REVIEW

File: CopyKeys/CopyKeys.ps1 [hunk 1]
```diff
@@ -7,11 +7,17 @@
...
+### <param name="AllowResourceMoverAccess">Switch parameter indicating if the MSI created by
+### Azure Resource Mover for moving the selected VM resources need to be given access to the
+### target BEK/KEK key vaults.</param>
 ### <param name="FilePath">Optional parameter defining the location of the output file.</param>
 ### ---------------------------------------------------------------
 
 [CmdletBinding()]
 param(
+    [Parameter(Mandatory = $false,
+               HelpMessage="Location of the output file.")]
```
Comment (ayfathim) [inline CopyKeys/CopyKeys.ps1:19]: Correct the comment

File: CopyKeys/CopyKeys.ps1 [hunk 2]
```diff
@@ -303,20 +309,31 @@ class Source
...
     static [string] $asrSuffix = "-asr"
     static [string] $authHeader = "authorization"
     static [string] $contentType = "application/json"
+    static [string] $httpGet = "GET"
     static [string] $httpPost = "POST"
     static [int] $keyVaultNameMaxLength = 24
     static [string] $loadingBEK = "Loading target BEK vault"
     static [string] $loadingKEK = "Loading target KEK vault"
     static [string] $loadingRG = "Loading resource groups"
+    static [string] $managementAzureEndpoint = "https://management.azure.com"
```
Comment (ayfathim) [inline CopyKeys/CopyKeys.ps1:322]: This azure endPoint does not work for fairfax region

**Review by ayfathim — CHANGES_REQUESTED** [review]

**Review by ayfathim — APPROVED** [review]

File: CopyKeys/CopyKeys.ps1 (hunk 2 above)
Comment (punit1396) [reply CopyKeys/CopyKeys.ps1]: Using the arm endpoint using context.environment

File: CopyKeys/CopyKeys.ps1 (hunk 1 above)
Comment (punit1396) [reply CopyKeys/CopyKeys.ps1]: corrected

**Pull request merged**